src/Campaigns/Repositories/CampaignRepository.php

```diff
@@ -0,0 +1,204 @@
+                    'long_desc' => $campaign->longDescription,
+                    'campaign_logo' => $campaign->logo,
+                    'campaign_image' => $campaign->image,
+                    'primary_color' => $campaign->primaryColor,
+                    'secondary_color' => $campaign->secondaryColor,
+                    'goal' => $campaign->goal,
+                    'status' => $campaign->status->getValue(),
+                    'start_date' => $campaign->startDate,
+                    'end_date' => $campaign->endDate,
+                    'date_created' => $dateCreatedFormatted
+                ]);
+
+            $campaignId = DB::last_insert_id();
+        } catch (Exception $exception) {
+            DB::query('ROLLBACK');
+
+            Log::error('Failed creating a campaign', compact('campaign'));
+
+            throw new $exception('Failed creating a campaign');
+        }
+
+        DB::query('COMMIT');
+
+        $campaign->id = $campaignId;
+        $campaign->createdAt = $dateCreated;
+
+        Hooks::doAction('givewp_campaign_created', $campaign);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @throws Exception|InvalidArgumentException
+     */
+    public function update(Campaign $campaign): void
+    {
+        $this->validateProperties($campaign);
+
+        Hooks::doAction('givewp_campaign_updating', $campaign);
+
+        DB::query('START TRANSACTION');
+
+        try {
+            DB::table('give_campaigns')
+                ->where('id', $campaign->id)
+                ->update([
+                    'campaign_title' => $campaign->title,
+                    'short_desc' => $campaign->shortDescription,
+                    'long_desc' => $campaign->longDescription,
+                    'campaign_logo' => $campaign->logo,
+                    'campaign_image' => $campaign->image,
+                    'primary_color' => $campaign->primaryColor,
+                    'secondary_color' => $campaign->secondaryColor,
+                    'goal' => $campaign->goal,
+                    'status' => $campaign->status->getValue(),
+                    'start_date' => $campaign->startDate,
+                    'end_date' => $campaign->endDate,
+                ]);
+        } catch (Exception $exception) {
+            DB::query('ROLLBACK');
+
+            Log::error('Failed updating a campaign', compact('campaign'));
+
+            throw new $exception('Failed updating a campaign');
+        }
+
+        DB::query('COMMIT');
+
+        Hooks::doAction('givewp_campaign_updated', $campaign);
+    }
+
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     */
+    public function delete(Campaign $campaign): bool
+    {
+        DB::query('START TRANSACTION');
+
+        Hooks::doAction('givewp_campaign_deleting', $campaign);
+
+        try {
+            DB::table('give_campaigns')
+                ->where('id', $campaign->id)
+                ->delete();
+
+            // todo: also delete campaign page?
```

Comment:
We'll probably want to hook into the `givewp_campaign_deleting` hook above for the campaign page to delete itself (which or may not exist).

Reply:
We can create an action later, for now, I will just update the todo tag.

Reply:
Resolved: https://github.com/impress-org/givewp/pull/7520/commits/10be69b385c1bc4a593dd8af60020915463c5762